caikit/core/model_management/factories.py

```diff
@@ -33,10 +39,25 @@
 model_finder_factory = ImportableFactory("ModelFinder")
 model_finder_factory.register(LocalModelFinder)
 model_finder_factory.register(MultiModelFinder)
+try:
+    # Local
+    from caikit.runtime.client import RemoteModelFinder
```

Comment:
This seems to always fail due to a circular import caused by [these imports in runtime.__init__](https://github.com/caikit/caikit/blob/e33e280c5d5bf3c6066543eabf2b2e328c793f48/caikit/runtime/__init__.py#L16). Can those be removed or are those used alot by services/examples

Reply:
Hm. I haven't gotten too deeply into the review, but I would really like to not have this here. The call to `model_finder_factory.register` can live in `caikit.runtime.client` directly (i.e. the new instance registers itself when created). This would have two nice benefits. The downside would be that a user would need to explicitly import `caikit.runtime.client` for the registration to work, but since `model_finder_factory` is an `ImportableFactory`, this could still be accomplished through config by setting `import_class: caikit.runtime.client.RemoteModelFinder`.

Reply:
👍 I keep forgetting that it's an importable factory. I'll push a change